packages/integration-tests/src/api/session.ts

```diff
@@ -51,112 +53,148 @@ export const consent = async (interactionCookie: string) =>
-      json: {
-        email,
-        code,
-      },
     })
     .json<RedirectResponse>();
+};
 
 export const sendSignInUserWithEmailPasscode = (email: string, interactionCookie: string) =>
-  api.post('session/sign-in/passwordless/email/send-passcode', {
+  api.post('session/passwordless/email/send', {
     headers: {
       cookie: interactionCookie,
     },
     json: {
       email,
+      flow: PasscodeType.SignIn,
     },
   });
 
-export const verifySignInUserWithEmailPasscode = (
+export const verifySignInUserWithEmailPasscode = async (
   email: string,
   code: string,
   interactionCookie: string
-) =>
-  api
-    .post('session/sign-in/passwordless/email/verify-passcode', {
+) => {
+  await api.post('session/passwordless/email/verify', {
+    headers: {
+      cookie: interactionCookie,
+    },
+    json: {
+      email,
+      code,
+      flow: PasscodeType.SignIn,
+    },
+  });
+
+  return api
+    .post('session/sign-in/passwordless/email', {
       headers: {
         cookie: interactionCookie,
       },
-      json: {
-        email,
-        code,
-      },
     })
     .json<RedirectResponse>();
+};
 
 export const sendRegisterUserWithSmsPasscode = (phone: string, interactionCookie: string) =>
-  api.post('session/register/passwordless/sms/send-passcode', {
+  api.post('session/passwordless/sms/send', {
     headers: {
       cookie: interactionCookie,
     },
     json: {
       phone,
+      flow: PasscodeType.Register,
     },
   });
 
-export const verifyRegisterUserWithSmsPasscode = (
+export const verifyRegisterUserWithSmsPasscode = async (
   phone: string,
   code: string,
   interactionCookie: string
-) =>
-  api
-    .post('session/register/passwordless/sms/verify-passcode', {
+) => {
+  await api.post('session/passwordless/sms/verify', {
+    headers: {
+      cookie: interactionCookie,
+    },
+    json: {
+      phone,
+      code,
+      flow: PasscodeType.Register,
+    },
+  });
+
+  return api
+    .post('session/register/passwordless/sms', {
       headers: {
         cookie: interactionCookie,
       },
-      json: {
-        phone,
-        code,
-      },
     })
     .json<RedirectResponse>();
+};
 
 export const sendSignInUserWithSmsPasscode = (phone: string, interactionCookie: string) =>
-  api.post('session/sign-in/passwordless/sms/send-passcode', {
+  api.post('session/passwordless/sms/send', {
     headers: {
       cookie: interactionCookie,
     },
     json: {
       phone,
+      flow: PasscodeType.SignIn,
     },
   });
 
-export const verifySignInUserWithSmsPasscode = (
+export const verifySignInUserWithSmsPasscode = async (
   phone: string,
   code: string,
   interactionCookie: string
-) =>
-  api
-    .post('session/sign-in/passwordless/sms/verify-passcode', {
+) => {
+  await api.post('session/passwordless/sms/verify', {
+    headers: {
+      cookie: interactionCookie,
+    },
+    json: {
+      phone,
+      code,
+      flow: PasscodeType.SignIn,
+    },
+  });
+
+  return api
```

Comment:
split into two steps would be more reasonable. Maybe split the test case as well, since those are API-specific integration tests.